Source/qol/stash.cpp

```diff
@@ -352,21 +352,17 @@ void DrawStash(const Surface &out)
 
 	constexpr Displacement offset { 0, INV_SLOT_SIZE_PX - 1 };
 
-	for (auto slot : StashGridRange) {
-		StashStruct::StashCell itemId = Stash.GetItemIdAtPosition(slot);
-		Item &item = Stash.stashList[itemId];
-		if (Stash.IsItemAtPosition(slot)) {
-			InvDrawSlotBack(out, GetStashSlotCoord(slot) + offset, InventorySlotSizeInPixels, item._iMagical);
-		}
-	}

```

Comment:
Stephen pointed out the rendering issue on items two cells wide which was the reason for the backs being drawn first. A minimally invasive fix would be to move the code that performs the stashList lookup inside the if statement. It's more relevant for the scope that `item` is actually used as well.

If you did want to fix the rendering for inventory items so backs can be drawn in one pass that'd be awesome :D, seems like a fair bit of work required though.

Reply:
I can take a look at batch rendering separately as that's likely something that may affect more than just the stash. :)